app/views/events/_below_table.html.haml

```diff
@@ -13,4 +13,88 @@
+              = f.label t('events_module.bulk_edit.set_new_track')
+              = f.hidden_field 'bulk_set_attribute', value: 'track_name', name: 'bulk_set_attribute'
+              = select_tag 'bulk_set_value', options_for_select(@conference.tracks.map { |t| t.name.to_s }), include_blank: t('select_one'), style: "width:400px"
+              = f.button :submit, 
+                         t('set'), 
+                         name: 'bulk_set', 
+                         id: 'bulk_set',
+                         data: { confirm: t('events_module.bulk_edit.set_track_confirm', count: @num_of_matching_events) }, 
+                         class: 'danger'
+          = simple_form_for(:bulk_edit_event_type, url: batch_actions_events_path(request.query_parameters), method: :post) do |f|
+            = f.label t('events_module.bulk_edit.set_new_event_type')
+            = f.hidden_field 'bulk_set_attribute', value: 'event_type', name: 'bulk_set_attribute'
+            = select_tag 'bulk_set_value', options_for_select(translated_options(@conference.allowed_event_types_as_list)), include_blank: t('select_one'), style: "width:400px"
+            = f.button :submit, 
+                       t('set'), 
+                       name: 'bulk_set', 
+                       id: 'bulk_set',
+                       data: { confirm: t('events_module.bulk_edit.set_events_type_confirm', count: @num_of_matching_events) }, 
+                       class: 'danger'
+          = simple_form_for(:bulk_edit_event_state, url: batch_actions_events_path(request.query_parameters), method: :post) do |f|
+            = f.label t('events_module.bulk_edit.set_new_state')
+            = f.hidden_field 'bulk_set_attribute', value: 'state', name: 'bulk_set_attribute'
+            = select_tag 'bulk_set_value',
+                         options_for_select(Event.state_machine.states.map { |st| [t(st.name.to_s, scope: 'conferences_module'), st.name.to_s] }), 
+                         include_blank: t('select_one'), style: "width:400px"
+            = f.button :submit, 
+                       t('set'), 
+                       name: 'bulk_set', 
+                       id: 'bulk_set',
+                       data: { confirm: t('events_module.bulk_edit.set_events_confirm', count: @num_of_matching_events) }, 
+                       class: 'danger'
+          = simple_form_for(:bulk_edit_add_person, 
+                            url: batch_actions_events_path(request.query_parameters), 
+                            method: :post, 
+                            data: { persons: Person.fullname_options } ) do |f|
+            .peoplefilter
+              = f.label t('events_module.bulk_edit.add_person')
+              = text_field_tag :filter, 'filter', style: "width:200px"
+              = select_tag :person_id, 
+                           options_for_select(Person.all.sort_by(&:full_name).map{|p| [p.full_name_annotated, p.id]}), 
```

Comment:
This is a really bad idea. I usually hate opening the "Edit persons" view on an event, because it takes ages to load and render, due to the thousands of users in frab. They need to be queried, put in the selector and the type ahead usually is rather slow as well. I've always wanted to make this into some sort of AJAX thingy querying the server on demand after three characters have been typed.

Having this huge list delivered on every event index, wether you want to bulk edit it or not, is simply unacceptably wasteful. It may work on test databases, but after managing more than one or two conferences in the same instance, this is annoying.

Reply:
I see where this is coming from. You want to set an assistance to all the matching lectures. I wonder if that can be better achieved by a modal overlay, also leaving more space.

I will then look into how to make the type ahead work better. Honestly, the current implementation is a kludge. You never see while typing ahead if you actually narrowed the list down already.